### DISCUSSION TITLE
Menu item shows in tree view even if page is unpublished

### DISCUSSION THREAD

**Discussion by smenzer** [requarks/wiki#6322] [Error / Bug Report]
I have a page that is unpublished, yet it still shows up in the navigation menu when using the tree nav version, even when not logged in at all.

I've set the page as unpublished in the page settings:
<img width="991" alt="image" src="https://user-images.githubusercontent.com/624392/230177104-2eadac24-147d-456e-ad1d-745f34ea7f2f.png">

I would not expect unpublished pages to show up in the menu at all, let alone when I'm viewing as a guest and not an admin.

**Comment by chaojian-zhang** [comment]
Having same issue with Wiki.js 2.5.307. One can take a look the example at our public wiki site: https://wiki.methodox.io/en/KnowledgeBase

![Example](https://github.com/user-attachments/assets/16cf8f57-d31b-430b-84fb-6504a7fb4344)

The **_GraphQL_** page is not published but it's showing up.

Opening it (https://wiki.methodox.io/en/KnowledgeBase/GraphQL) will show this page:

![Error](https://github.com/user-attachments/assets/914d5c90-7ee3-4a68-96c1-5b94a30f31a9)

The visitor is on **Guest** group:

![Group Settings](https://github.com/user-attachments/assets/d5708523-1511-473f-b37e-8913c5e98203)

![Rule Setting](https://github.com/user-attachments/assets/97122e20-e78e-4794-9af6-f617b098c3c4)

![Rule Setting 2](https://github.com/user-attachments/assets/00d02185-f808-497c-9a75-6acc88a94e59)

From the admin side, the page is set as unpublished:

![Page Setting](https://github.com/user-attachments/assets/11b94d6b-71e3-42f8-b64c-042ac3f9d1c6)

And the page itself shows it:

![Page View](https://github.com/user-attachments/assets/abf0012d-a79d-4a87-ae97-1a6ecc54c7e4)
